ddtrace/contrib/botocore/patch.py

```diff
@@ -481,225 +123,104 @@ def patched_api_call(original_func, instance, args, kwargs):
-                                    )
-                            elif (
-                                "MessageAttributes" in message
-                                and "_datadog" in message["MessageAttributes"]
-                                and "StringValue" in message["MessageAttributes"]["_datadog"]
-                            ):
-                                # The message originated from SQS
-                                message_body = message
-                                context_json = json.loads(message_body["MessageAttributes"]["_datadog"]["StringValue"])
-                            elif (
-                                "MessageAttributes" in message
-                                and "_datadog" in message["MessageAttributes"]
-                                and "BinaryValue" in message["MessageAttributes"]["_datadog"]
-                            ):
-                                # Raw message delivery
-                                message_body = message
-                                context_json = json.loads(
-                                    message_body["MessageAttributes"]["_datadog"]["BinaryValue"].decode()
-                                )
-                            else:
-                                log.debug("DataStreams did not handle message: %r", message)
-
-                            pathway = context_json.get(PROPAGATION_KEY_BASE_64, None) if context_json else None
-                            ctx = pin.tracer.data_streams_processor.decode_pathway_b64(pathway)
-                            ctx.set_checkpoint(["direction:in", "topic:" + queue_name, "type:sqs"])
-
-                except Exception:
-                    log.debug("Error receiving SQS message with data streams monitoring enabled", exc_info=True)
 
-                return result
-
-            else:
-                result = original_func(*args, **kwargs)
-
-                if endpoint_name == "kinesis" and operation == "GetRecords" and config._data_streams_enabled:
+    operation = None
+    params = None
+    if args:
+        operation = get_argument_value(args, kwargs, 0, "operation_name")
+        params = get_argument_value(args, kwargs, 1, "api_params")
+
+    function_vars = {
+        "endpoint_name": endpoint_name,
+        "operation": operation,
+        "params": params,
+        "pin": pin,
+        "trace_operation": trace_operation,
+    }
+
+    if endpoint_name == "kinesis":
+        return patched_kinesis_api_call(
+            original_func=original_func,
+            instance=instance,
+            args=args,
+            kwargs=kwargs,
+            function_vars=function_vars,
+        )
+    elif endpoint_name == "sqs":
+        return patched_sqs_api_call(
+            original_func=original_func,
+            instance=instance,
+            args=args,
+            kwargs=kwargs,
+            function_vars=function_vars,
+        )
+    else:
+        with pin.tracer.trace(
```

Comment:
Take the code below this and move it into a separate method, maybe "patch_fallthrough" (or something more obvious/clever) so that the `patch()` method is only handling the switching between patching mechanisms.

Reply:
Done!